include/nlohmann/json.hpp

```diff
@@ -3678,9 +3685,15 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
     with range checking
     @sa see @ref value() for access by value with a default value
 
-    @since version 1.0.0
+    @since version 1.0.0; template added in version 3.10.0
     */
-    reference operator[](const typename object_t::key_type& key)
+    template < class KeyT, typename std::enable_if <
+                   !std::is_same<typename std::decay<KeyT>::type, json_pointer>::value&&
+#if defined(JSON_HAS_CPP_17)
+                   !std::is_same<typename std::decay<KeyT>::type, std::string_view>::value&&
```

Comment:
It's not clear to me why you need to special case this for std::string_view ?

Reply:
Good catch. I'll refactor.